include/zephyr/net/coap_client.h

```diff
@@ -137,6 +137,16 @@ int coap_client_init(struct coap_client *client, const char *info);
 int coap_client_req(struct coap_client *client, int sock, const struct sockaddr *addr,
 		    struct coap_client_request *req, struct coap_transmission_parameters *params);
 
+/**
+ * @brief Cancel CoAP Requests
+ *
+ * When network connectivity is lost, terminate all pending requests.
+ *
+ * @param client Client instance.
+ * @return zero when operation started successfully or negative error code otherwise.
+ */
+int coap_client_cancel_req(struct coap_client *client);
```

Comment:
Do you have to call this from within the callback, or outside, or does it matter?
Is it safe to call from callback?

Reply:
Good question. I will devise a test for this.

Reply:
Good question. I am making a small change so it should be ok to call it either way.